### PR TITLE
AVRO-3523: Move contributions guidelines to website

### PR DESCRIPTION
**Pull request by clesaec**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Avro Jira](https://issues.apache.org/jira/browse/AVRO-3523) issues and references them in the PR title. For example, "AVRO-1234: My Avro PR"
  - https://issues.apache.org/jira/browse/AVRO-XXX
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason: For doc.

### Commits

- [x] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](https://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

File: doc/content/en/project/How to contribute/_index.md [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+<details><summary>Ruby</summary>
+
+```shell
+cd avro-trunk/lang/ruby
+gem install echoe
+rake clean test
+```
+</details>
+
+<details><summary>PHP</summary>
```
Comment (martin-g) [inline doc/content/en/project/How to contribute/_index.md:129]: Please add an entry for `lang/rust` too.

File: doc/content/en/project/How to contribute/_index.md [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+```shell
+cd avro-trunk/lang/py
+ant clean test
+```
+</details>
+
+<details><summary>Python3</summary>
+
+```shell
+cd avro-trunk/lang/py3
```
[the remark this answers is not in the document]
Comment (clesaec) [reply doc/content/en/project/How to contribute/_index.md]: Indeed, there is only one python section now, remove python 3 section.
Only python section with 
`./setup.py build test`
for build

File: doc/content/en/project/How to contribute/_index.md [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+cd avro-trunk/lang/java
+mvn clean test
+```
+</details>
+
+<details><summary>Python</summary>
+
+```shell
+cd avro-trunk/lang/py
+ant clean test
```
[the remark this answers is not in the document]
Comment (clesaec) [reply doc/content/en/project/How to contribute/_index.md]: As `./build.sh test` doesn't work for me in local, i put `./setup.py build test`

File: doc/content/en/project/How to contribute/_index.md (hunk 1 above)
Comment (clesaec) [reply doc/content/en/project/How to contribute/_index.md:129]: :+1:  rust section added

File: doc/content/en/project/How to contribute/_index.md [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+Click on it, fill Conversation form, and create it.
+Link this PR to the corresponding JIRA ticket (on JIRA ticket, add PR to "Issue Links" chapter, and add label 'pull-request-available' to it .
+
+
+<details><summary><b>Via Patch</b> (if you don't have github account)</summary>
+<blockquote>
+<details><summary><b>Clone avro repository</b></summary> 
+
+```shell
+git clone https://github.com/apache/avro.git
```
[the remark this answers is not in the document]
Comment (clesaec) [reply doc/content/en/project/How to contribute/_index.md]: added

File: doc/content/en/project/How to contribute/_index.md (hunk 1 above)
Comment (KalleOlaviNiemitalo) [reply doc/content/en/project/How to contribute/_index.md:129]: There is also `lang/csharp/build.sh` for C#. When I tried it with Avro 1.11.0 on Windows, Avro.Test.LogicalTypeTests.TestDecimal failed with FormatException, but that has already been fixed as part of AVRO-3468.

Comment (clesaec) [reply doc/content/en/project/How to contribute/_index.md:129]: So, nothing to do on doc ?

Comment (clesaec) [reply doc/content/en/project/How to contribute/_index.md:137]: Yes, sorry, i didn't see i forget C# section, it's added

**Review by opwvhk — APPROVED** [review]

File: doc/content/en/project/How to contribute/_index.md [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+
+Apply the patch uploaded by the user or check out their pull request. Edit the CHANGES.txt file, adding a description of the change, including the bug number it fixes. Add it to the appropriate section - BUGFIXES, IMPROVEMENTS, NEW FEATURES. Please follow the format in CHANGES.txt file. While adding an entry please add it to the end of a section. Use the same entry for the first line of the git commit message.
+
+Changes are normally committed to master first, then, if they're backward-compatible, cherry-picked to a branch.
+
+When you commit a change, resolve the issue in Jira. When resolving, always set the fix version and assign the issue. Set the fix version to either to the next minor release if the change is compatible and will be merged to that branch, or to the next major release if the change is incompatible and will only be committed to trunk. Assign the issue to the primary author of the patch. If the author is not in the list of project contributors, edit their Jira roles and make them an Avro contributor.
+</details>
+</blockquote>
+</details>
+
```
Comment (RyanSkraba) [inline doc/content/en/project/How to contribute/_index.md:280]:
```suggestion
-->
```

File: doc/content/en/project/How to contribute/_index.md [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+```shell
+git commit -m "AVRO-1234: Fix NPE by adding check to ..."
+git push
+```
+On your github fork site, a button will propose you to build the Pull Request.
+Click on it, fill Conversation form, and create it.
+Link this PR to the corresponding JIRA ticket (on JIRA ticket, add PR to "Issue Links" chapter, and add label 'pull-request-available' to it .
+
+
+<details><summary><b>Via Patch</b> (if you don't have github account)</summary>
```
Comment (RyanSkraba) [inline doc/content/en/project/How to contribute/_index.md:196]:
```suggestion
<!--  TODO: Fix formatting and decide whether this should be included in this section, moved to another section
<details><summary><b>Via Patch</b> (if you don't have github account)</summary>
```

File: doc/content/en/project/How to contribute/_index.md [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+<details><summary>PHP</summary>
+
+```shell
+cd avro-trunk/lang/php
+./build.sh clean
+./build.sh test
+```
+</details>
+
+<details><summary>Documentation</summary>
```
Comment (RyanSkraba) [inline doc/content/en/project/How to contribute/_index.md:146]: This is largely overpromising what the given commands do!  I'd just take this out.

**Review by RyanSkraba — CHANGES_REQUESTED** [review]
Hello!  Thanks so much for this work and your patience.

There's a lot of duplication with the materials in `content/endocs/next/Contribution Guidelines/_index.html`, and this conflicts and/or duplicates information there.

This is consistent with the source material in the wiki -- it's not _great_ because a lot of the source material is out of date or inconsistent or just not extremely useful!

I appreciate that you did some "information" hiding with the `<details><summary> ...` sections but (especially in the via a patch part), a lot of the formatting is just wrong when it's wrapped in details!

**This is what I suggest** -- let's take out the patch section entirely (via a comment), and merge the rest, and drop the duplicate page (for now).  Once this is present on the website, we can probably expect many improvements!

File: doc/content/en/project/How to contribute/_index.md (hunk 6 above)
Comment (clesaec) [reply doc/content/en/project/How to contribute/_index.md:183]: section put in comments

File: doc/content/en/project/How to contribute/_index.md (hunk 7 above)
Comment (clesaec) [reply doc/content/en/project/How to contribute/_index.md]: Documentation generation removed.

**Comment by clesaec** [comment]
Hi Ryan,
Thanks for advices, i made the corresponding change.

**Review by RyanSkraba — APPROVED** [review]
LGTM!  Thanks for the contribution!

**Pull request merged**